### PR TITLE
[DOCS] Enrich docs for Kyuubi Hive JDBC driver

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

Update the outdated words for Kyuubi Hive JDBC driver, and supply more details about Kerberos authentication.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

image

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: docs/quick_start/quick_start_with_jdbc.md [hunk 1]
```diff
@@ -15,82 +15,76 @@
...
+        String clientKeytab = args[1];    // Keytab file location
+        String serverPrincipal = arg[2];  // Kerberos principal used by Kyuubi Server
+        String kyuubiJdbcUrl = String.format(kyuubiJdbcUrl, clientPrincipal, clientKeytab, serverPrincipal);
+        Connection conn = DriverManager.getConnection(kyuubiJdbcUrl);
+        Statement stmt = conn.createStatement();
+        ResultSet rs = st.executeQuery("show databases");
+        while (rs.next()) {
+            System.out.println(rs.getString(1));
+        }
+        rs.close();
```
Comment (yaooqinn) [inline docs/quick_start/quick_start_with_jdbc.md:85]: nit: better in `finally` and wrap each close with separated `try-finally` blocks

File: docs/quick_start/quick_start_with_jdbc.md [hunk 2]
```diff
@@ -15,82 +15,76 @@
...
-        st.close();
+        rs.close();
+        stmt.close();
         conn.close();
     }
 }
 ```
+## Connect to Kerberized Kyuubi Server
+
+The following Java code is using a keytab file to login and connect to Kyuubi Server by JDBC.
```
Comment (yaooqinn) [inline docs/quick_start/quick_start_with_jdbc.md:65]:
```suggestion
The following Java code uses a keytab file to login and connect to Kyuubi Server by JDBC.
```

**Review by bowenliang123 — APPROVED** [review]
LGTM

File: docs/quick_start/quick_start_with_jdbc.md (hunk 1 above)
Comment (pan3793) [reply docs/quick_start/quick_start_with_jdbc.md]: updated

**Review by yaooqinn — APPROVED** [review]

**Pull request closed without merging**